### DISCUSSION TITLE
Deploy NextJS with 2000 static pages

### DISCUSSION THREAD

**Discussion by prasanthbattula** [vercel/next.js#28294] [Help] [answered]
I have developed a Next JS app which has more than **2000 static pages**. These pages need to be generated at build time. But it is taking very long for building. I tried to deploy this app on Vercel. But it failed due to long build time (more than 45 minutes).

**Comment by delbaoliveira** [comment, marked as answer]
Hey @prasanthbattula!

Could you give more context on how you're fetching the data for your pages? 

Alternatively, if you prefer static, you can use [incremental static regeneration](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) to build a subset of your pages at build-time and the rest at runtime. Here's an in-depth [article](https://www.smashingmagazine.com/2021/04/incremental-static-regeneration-nextjs/#incremental-static-regeneration-isr).

Saying that, 45 mins sounds long for 2,000 pages. Is there anything you can do to optimize the function that fetches your page data? E.g. Is your database close to your server?

**Reply by prasanthbattula** [reply]
Thank you
